src/main/java/org/sagebionetworks/web/client/widget/entity/file/FileDownloadMenuItem.java

```diff
@@ -138,22 +138,19 @@ public FileHandle getFileHandle() {
 		return null;
 	}
 
+	
 	public String getDirectDownloadURL(FileEntity fileEntity, FileHandle fileHandle, String fileNameOverride) {
 		String externalUrl = null;
 		if (fileHandle instanceof ExternalFileHandle) {
 			externalUrl = ((ExternalFileHandle) fileHandle).getExternalURL();
 		}
 
 		String directDownloadURL = null;
-		if (externalUrl == null)
+		if (externalUrl != null && externalUrl.toLowerCase().startsWith(WebConstants.SFTP_PREFIX)) {
+			// point to sftp proxy instead
+			directDownloadURL = Uploader.getSftpProxyLink(fileNameOverride, externalUrl, ginInjector.getSynapseProperties(), gwt);
+		} else {
 			directDownloadURL = jsniUtils.getFileHandleAssociationUrl(fileEntity.getId(), FileHandleAssociateType.FileEntity, fileHandle.getId());
```

Comment:
For everything else (including external URLs), go through the FHA servlet to get the presigned URL, and register it as a download.